From 2c48579454f7cd49500cb03ae23973c6880ea5ea Mon Sep 17 00:00:00 2001
Message-Id: <2c48579454f7cd49500cb03ae23973c6880ea5ea@dist-git>
From: Eric Blake <eblake@redhat.com>
Date: Tue, 7 Oct 2014 17:06:17 -0600
Subject: [PATCH] RHEL: qemu: support relative backing for RHEL 7.0.z qemu

RHEL-only: https://bugzilla.redhat.com/show_bug.cgi?id=1150322

qemu-kvm-rhev for RHEL 7.0.z backported enough code to allow
relative backing file manipulations, but could not backport
everything from upstream.  So, instead of providing the upstream
'change-backing-file' QMP command, it added a downstream-only
'__com.redhat_change-backing-file' as a witness that relative
backing is supported, but not as full-featured.  Since libvirt
from RHEL 7.1 may be driving an older qemu, we need to be able
to recognize the alternate spelling.

* src/qemu/qemu_capabilities.c (virQEMUCapsCommands): Also
recognize downstream spelling.

Signed-off-by: Eric Blake <eblake@redhat.com>
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/qemu/qemu_capabilities.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index 28d5321..206133d 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -1464,6 +1464,7 @@ struct virQEMUCapsStringFlags virQEMUCapsCommands[] = {
     { "add-fd", QEMU_CAPS_ADD_FD },
     { "nbd-server-start", QEMU_CAPS_NBD_SERVER },
     { "change-backing-file", QEMU_CAPS_CHANGE_BACKING_FILE },
+    { "__com.redhat_change-backing-file", QEMU_CAPS_CHANGE_BACKING_FILE },
     { "rtc-reset-reinjection", QEMU_CAPS_RTC_RESET_REINJECTION },
     { "migrate-incoming", QEMU_CAPS_INCOMING_DEFER },
 };
-- 
2.9.0

